File: persons-finder-projects/java/persons-finder-4j/src/main/java/com/persons/finder/data/IndexField.java
package com.persons.finder.data;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@MappedSuperclass
@Data
public class IndexField {

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	@Getter @Setter private Long id;
	
	public IndexField() {
		
	}
	
	public IndexField(Long id) {
		this.id = id;
	}
}

File: persons-finder-projects/kotlin/persons-finder/src/main/kotlin/com/persons/finder/data/IndexField.kt
package com.persons.finder.data

import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
open class IndexField {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    constructor() {
    }

    constructor(id: Long?) {
        this.id = id
    }
}